modules/core/history.js

```diff
@@ -158,11 +159,12 @@ export function coreHistory(context) {
                         };
                     })
                     .on('start', function() {
-                        _perform([action0], 0);
+                        resolve(_perform([action0], 0));
```

Comment:
instead of waiting x milliseconds, which is error-prone, returning a promise makes this test more deterministic